mountpoint-s3/tests/fuse_tests/cache_test.rs

```diff
@@ -1,19 +1,83 @@
 use crate::common::cache::CacheTestWrapper;
-use crate::common::fuse::create_fuse_session;
-use crate::common::fuse::s3_session::create_crt_client;
-use crate::common::s3::{get_express_bucket, get_standard_bucket, get_test_bucket_and_prefix};
-use mountpoint_s3::data_cache::{DataCache, DiskDataCache, DiskDataCacheConfig, ExpressDataCache};
+use crate::common::fuse::s3_session::{create_crt_client, create_test_client};
+use crate::common::fuse::{create_fuse_session, TestClient};
+use crate::common::s3::{get_express_bucket, get_standard_bucket, get_test_prefix, get_test_region};
+use fuser::BackgroundSession;
+use mountpoint_s3::data_cache::{
+    build_prefix, get_s3_key, BlockIndex, DataCache, DiskDataCache, DiskDataCacheConfig, ExpressDataCache,
+};
+use mountpoint_s3::object::ObjectId;
 use mountpoint_s3::prefetch::caching_prefetch;
+use mountpoint_s3_client::types::{PutObjectParams, PutObjectTrailingChecksums};
 use mountpoint_s3_client::S3CrtClient;
 use rand::{Rng, RngCore, SeedableRng};
 use rand_chacha::ChaChaRng;
 use std::fs;
 use std::time::Duration;
+use tempfile::TempDir;
 use test_case::test_case;
 
 const CACHE_BLOCK_SIZE: u64 = 1024 * 1024;
 const CLIENT_PART_SIZE: usize = 8 * 1024 * 1024;
 
+/// A test that checks that an invalid block may not be served from the cache
+#[test]
+fn express_invalid_block_read() {
+    let region = get_test_region();
+    let bucket = get_standard_bucket();
+    let cache_bucket = get_express_bucket();
+    let prefix = get_test_prefix("express_invalid_block_read");
+
+    // Mount the bucket
+    let client = create_crt_client(CLIENT_PART_SIZE, CLIENT_PART_SIZE);
+    let cache = CacheTestWrapper::new(ExpressDataCache::new(
+        client.clone(),
+        Default::default(),
+        &bucket,
+        &cache_bucket,
+    ));
+    let (mount_point, _session) = mount_bucket(client, cache.clone(), &bucket, &prefix);
+
+    // Put an object to the mounted bucket
+    let bucket_client = create_test_client(&region, &bucket, &prefix);
```

Comment:
Is there a reason to use `TestClient` here? Why not just use an S3 client directly? This test would be much simpler and no need to add the `get_object_etag` method.